jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -785,6 +836,12 @@ Object runNativeMethod(Addressable downcallAddr, SegmentAllocator segmtAllocator
 		/*[ENDIF] JAVA_SPEC_VERSION >= 21 */
 
 		long returnVal;
+		/*[IF JAVA_SPEC_VERSION >= 22]*/
+		Thread currentThread = Thread.currentThread();
+		HeapArgInfo heapArgInfo = threadHeapArgInfoMap.get(currentThread);
+		Object[] heapBases = (heapArgInfo != null) ? heapArgInfo.bases() : null;
+		long[] heapOffsets = (heapArgInfo != null) ? heapArgInfo.offsets() : null;
```

Comment:
I don't think locals for `heapBases` and `heapOffsets` are worthwhile
```java
		HeapArgInfo info = heapArgInfo.get();
```

Reply:
Will update all the code around related to `ThreadLocal`.

Reply:
Updated the code with `ThreadLocal` and verified in tests.